plugins/source/azure/resources/services/network/security_groups.go

```diff
@@ -1,527 +1,113 @@
-					},
-					{
-						Name:        "direction",
-						Description: "The direction of the rule The direction specifies if rule will be evaluated on incoming or outgoing traffic Possible values include: 'SecurityRuleDirectionInbound', 'SecurityRuleDirectionOutbound'",
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("SecurityRulePropertiesFormat.Direction"),
-					},
-					{
-						Name:        "provisioning_state",
-						Description: "The provisioning state of the security rule resource Possible values include: 'Succeeded', 'Updating', 'Deleting', 'Failed'",
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("SecurityRulePropertiesFormat.ProvisioningState"),
-					},
-					{
-						Name:        "name",
-						Description: "The name of the resource that is unique within a resource group This name can be used to access the resource",
-						Type:        schema.TypeString,
-					},
-					{
-						Name:        "etag",
-						Description: "A unique read-only string that changes whenever the resource is updated",
-						Type:        schema.TypeString,
-					},
-					{
-						Name:        "type",
-						Description: "The type of the resource",
-						Type:        schema.TypeString,
-					},
-					{
-						Name:        "id",
-						Description: "Resource ID",
-						Type:        schema.TypeString,
-						Resolver:    schema.PathResolver("ID"),
-					},
-				},
+				Name:     "type",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("Type"),
+			},
+			{
+				Name:     "location",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("Location"),
+			},
+			{
+				Name:     "tags",
+				Type:     schema.TypeJSON,
+				Resolver: schema.PathResolver("Tags"),
 			},
 		},
 	}
 }
 
-// ====================================================================================================================
-//
-//	Table Resolver Functions
-//
-// ====================================================================================================================
 func fetchNetworkSecurityGroups(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
 	svc := meta.(*client.Client).Services().Network.SecurityGroups
+
 	response, err := svc.ListAll(ctx)
+
 	if err != nil {
-		return diag.WrapError(err)
+		return errors.WithStack(err)
 	}
+
 	for response.NotDone() {
 		res <- response.Values()
 		if err := response.NextWithContext(ctx); err != nil {
-			return diag.WrapError(err)
+			return errors.WithStack(err)
 		}
 	}
-	return nil
-}
-func fetchNetworkSecurityGroupSecurityRules(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
-	p := parent.Item.(network.SecurityGroup)
-	res <- *p.SecurityRules
-	return nil
-}
-func fetchNetworkSecurityGroupFlowLogs(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
```

Comment:
Moved to `plugins/source/azure/resources/services/network/flow_logs.go`.
There's a list API to get the flow logs and based on it flow logs should a relation of watchers